### PR TITLE
poke: improvements to conversation view for debugging

### PR DESCRIPTION
**Pull request by spolu**
## Description

- Adds possibility to download content fragment text
- Improves poke conversation view with links to trace-converstaions, assistant poke pages and content fragment content

Screenshot from 2024-06-25 10-36-01

## Risk

N/A (no new page or endpoint added so security is already covered)

## Deploy Plan

- deploy `front`

### CODE REVIEW

File: front/lib/resources/content_fragment_resource.ts [hunk 1]
```diff
@@ -141,7 +142,29 @@ export class ContentFragmentResource extends BaseResource<ContentFragmentModel>
...
   }
 
-  renderFromMessage(message: Message): ContentFragmentType {
+  renderFromMessage({
+    auth,
+    conversationId,
+    message,
+  }: {
+    auth: Authenticator;
+    conversationId: string;
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/lib/resources/content_fragment_resource.ts:151]: message.conversationId is a number  no?

File: front/lib/api/assistant/messages.ts [hunk 2]
```diff
@@ -382,11 +389,68 @@ export async function fetchConversationMessages(
...
+  );
 
   return new Ok({
     hasMore,
     lastValue: renderedMessages.at(0)?.rank ?? null,
     messages: renderedMessages,
   });
 }
+
+export async function getAgentMessageRunIds(
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/lib/api/assistant/messages.ts]: This was removed

File: front/lib/resources/content_fragment_resource.ts [hunk 3]
```diff
@@ -150,6 +173,7 @@ export class ContentFragmentResource extends BaseResource<ContentFragmentModel>
       visibility: message.visibility,
       version: message.version,
       sourceUrl: this.sourceUrl,
+      textUrl: location.downloadUrl,
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/lib/resources/content_fragment_resource.ts:176]: Not sure I understand your question?

File: front/lib/resources/content_fragment_resource.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (flvndvd) [reply front/lib/resources/content_fragment_resource.ts:151]: 🤦

File: front/lib/resources/content_fragment_resource.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (flvndvd) [reply front/lib/resources/content_fragment_resource.ts:176]: Ok forgot about this, missed the line below.

File: front/lib/resources/content_fragment_resource.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (flvndvd) [reply front/lib/resources/content_fragment_resource.ts:151]: My point is that, even though the impact is minimal if we pass the wrong conversation ID in this case, we should avoid using different IDs for linked resources, such as message and conversation. This pattern seems pretty error-prone.

**Review by flvndvd — APPROVED** [review]

**Pull request merged**